src/main/java/rx/internal/schedulers/NewThreadWorker.java

```diff
@@ -112,7 +124,7 @@ static void purgeExecutors() {
      * @return true if the policy was successfully enabled 
      */
     public static boolean tryEnableCancelPolicy(ScheduledExecutorService exec) {
-        if (!PURGE_FORCE) {
+        if (!PURGE_FORCE && (!IS_ANDROID || IS_ANDROID_AND_SUPPORTS_SET_REMOVE_ON_CANCEL_POLICY)) {
```

Comment:
What i'm saying is, just make this 
```java
if (!PURGE_FORCE && IS_ANDROID_AND_SUPPORTS_SET_REMOVE_ON_CANCEL_POLICY)
```
to avoid the redundancy of having both variables and checking the same condition twice on the true case.